### ISSUE TITLE
Kernel panic after running for days

### ISSUE THREAD

**Issue by ThomasManthey** [Motion-Project/motion#963]
0. Reviewed guide and contributing documents? (Yes):
1. version [4.2.2] on RaspberryPiZero running Linux raspberryzerocam 4.9.35+ #1014 Fri Jun 30 14:34:49 BST 2017 armv6l GNU/Linux
2. installed as compiled from sources[ git]: 
3. standalone [motion]: 
4. video stream source [V4L, mmal]: 
5. hardware [ARM]: Broadcom BCM2835 CPU: ARMv6-compatible processor [410fb767] revision 7 (ARMv7), cr=00c5387d
6. operating system [Linux]: Linux raspberryzerocam 4.9.35+ #1014 Fri Jun 30 14:34:49 BST 2017 armv6l GNU/Linux
[dmesg]: 3.841204] This architecture does not have kernel memory protection
maybe a problem with wrong memory acess

**Comment by tosiara** [comment]
Do you have the stack trace?

**Comment by ThomasManthey** [comment]
Because of the kernel crash I only have made a photo from to days crash screen:
![IMG_20190622_172031](https://user-images.githubusercontent.com/52050946/59967434-f057e280-9529-11e9-91c9-114bfacebb95.jpg)
It was running round about 2 days.

**Comment by Mr-DaveDev** [comment]
What leads you to believe that this is caused by Motion?

**Comment by amitkriit** [comment]
This looks like an issue related to your WIFI/USB hardware/driver on the Raspberry Pi SBC. May be an USB device is disconnecting-reconnecting intermittently. A user mode application cannot cause a Kernel Panic unless something is wrong with the kernel itself.

**Comment by ThomasManthey** [comment]
Firstly:
Without running motion the pi runs for moths with camera and without any problems.


Secondly:

**A user mode application cannot cause a Kernel Panic unless something is wrong with the kernel itself.**
in normal cases you are right, but the message a startup **"[dmesg]: 3.841204] This architecture does not have kernel memory protection"** let me assume that in this case it can.

**Comment by tosiara** [comment]
Can you please upload your full motion log and syslog from the boot till crash?

**Comment by ThomasManthey** [comment]
It seems that the available files do not represent the actual crash. A motion-logfile I can't find. Is it necessary so set it up in motion config?

[kern.log.2.gz](https://github.com/Motion-Project/motion/files/3336011/kern.log.2.gz)
[kern.log.3.gz](https://github.com/Motion-Project/motion/files/3336012/kern.log.3.gz)
[kern.log.4.gz](https://github.com/Motion-Project/motion/files/3336013/kern.log.4.gz)
[messages.2.gz](https://github.com/Motion-Project/motion/files/3336014/messages.2.gz)
[messages.3.gz](https://github.com/Motion-Project/motion/files/3336015/messages.3.gz)
[messages.4.gz](https://github.com/Motion-Project/motion/files/3336016/messages.4.gz)


[syslog.2.gz](https://github.com/Motion-Project/motion/files/3336017/syslog.2.gz)
[syslog.3.gz](https://github.com/Motion-Project/motion/files/3336018/syslog.3.gz)
[syslog.4.gz](https://github.com/Motion-Project/motion/files/3336019/syslog.4.gz)
[syslog.5.gz](https://github.com/Motion-Project/motion/files/3336020/syslog.5.gz)
[syslog.6.gz](https://github.com/Motion-Project/motion/files/3336021/syslog.6.gz)
[syslog.7.gz](https://github.com/Motion-Project/motion/files/3336022/syslog.7.gz)

**Comment by tosiara** [comment]
It is in your config: https://motion-project.github.io/motion_config.html#log_file
Please reproduce the issue and upload all the corresponding logs together

**Comment by tosiara** [comment]
So far what most likely causes this crash is `uv4l_raspicam` driver:
```
Jun 20 19:17:14 raspberryzerocam uv4l_raspicam[124]: Starting UV4L Raspberry CSI Camera Driver: uv4l
Jun 20 19:17:14 raspberryzerocam uv4l_raspicam[124]: <notice> [core] Trying to load driver 'raspicam' from built-in drivers first...
Jun 20 19:17:14 raspberryzerocam uv4l_raspicam[124]: <warning> [core] Driver 'raspicam' not found
Jun 20 19:17:14 raspberryzerocam uv4l_raspicam[124]: <notice> [core] Trying to load driver 'raspicam' from external plug-in's instead...
Jun 20 19:17:14 raspberryzerocam uv4l_raspicam[124]: <notice> [driver] Dual Raspicam & TC358743 Video4Linux2 Driver v1.9.59 built Nov 11 2017
Jun 20 19:17:14 raspberryzerocam uv4l_raspicam[124]: <notice> [driver] Detected camera ov5647
Jun 20 19:17:14 raspberryzerocam uv4l_raspicam[124]: <notice> [driver] Selected format: 1920x1080, encoding: mjpeg, JPEG Video Capture
Jun 20 19:17:14 raspberryzerocam uv4l_raspicam[124]: <notice> [driver] Framerate max. 30 fps
Jun 20 19:17:14 raspberryzerocam uv4l_raspicam[124]: <notice> [driver] ROI: 0, 0, 1, 1
Jun 20 19:17:14 raspberryzerocam uv4l_raspicam[124]: <notice> [core] Device detected!
Jun 20 19:17:14 raspberryzerocam systemd[1]: Started LSB: Userspace Camera Driver.
Jun 20 19:17:14 raspberryzerocam uv4l_raspicam[124]: <notice> [core] Registering device node /dev/video0
```

**Comment by ThomasManthey** [comment]
Ok, I can do this.
Would you agree to reproduce the steps on another pi with memory protection? The log files and information's then should be available, because the kernel will terminate the application with memory protection exception.

**Comment by ThomasManthey** [comment]
Because this crash was after a very short run, here are my log files:
[log.tar.zip](https://github.com/Motion-Project/motion/files/3342728/log.tar.zip)

**Comment by tosiara** [comment]
Can you now disconnect rpi camera module and repeat the same test with a mjpeg net camera?
For example with some public URL, like: https://webcam1.lpl.org/axis-cgi/mjpg/video.cgi

**Comment by ThomasManthey** [comment]
I can try this. But how to?
Could you please shortly describe me the configuration steps?
I'm thinking about to run motion anyway on a different pi then the "camera-pi".

**Comment by tosiara** [comment]
Just put `netcam_url https://webcam1.lpl.org/axis-cgi/mjpg/video.cgi` into config

**Comment by ThomasManthey** [comment]
Ok, this dose not work for me, but at the mean time I try to get a stream using "mjpeg-streamer". The result was the same so I will make a bug report for the camera-driver. Many thanks for your support!

**Comment by ThomasManthey** [comment]
Because I hate after hours of searching to find a thread facing the same problem I have, but without a final solution, I want to describe the steps I've done to make video stable on my raspberry zero for those who are facing the same problem and finding this:
1. I updated the video driver to the newest version following this "https://raspberrypi.stackexchange.com/questions/22372/using-the-uv4l-driver-to-stream-video-from-the-raspberry-pi-camera-to-android-ph" description. Result: Crash after 1 hour
2. I updates the firmware of my raspberry zero using the firmware update tool raspi-update Result: a little bit more stable, but crash after one day
3. I compiled and installed another video driver "v4l2" using "https://www.linux-projects.org/uv4l/tutorials/real-time-object-detection/" Result: there is no crash anymore since 2 days
